### PR TITLE
Add the ability to add licenses tags to generated BUILD files.

### PR DESCRIPTION
**Pull request by paladine**
https://docs.bazel.build/versions/master/be/functions.html#licenses

If you put your BUILD files under third_party, bazel will throw an error
if there is no license tag.

Licenses are defined in the options section, as a Set[String]. It's a
global setting for now.

### CODE REVIEW

**Comment by paladine** [comment]
Updated the pull request - forgot to update tests :(

File: test/scala/com/github/johnynek/bazel_deps/ModelGenerators.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ object ModelGenerators {
     heads <- Gen.option(Gen.listOf(Gen.identifier))
     cache <- Gen.option(Gen.oneOf(ResolverCache.Local, ResolverCache.BazelOutputBase))
     prefix <- Gen.option(Gen.identifier.map(NamePrefix(_)))
-  } yield Options(vcp, dir, langs, res, trans, heads, cache, prefix)
+  } yield Options(vcp, dir, langs, res, trans, heads, cache, prefix, None)
```
Comment (johnynek) [inline test/scala/com/github/johnynek/bazel_deps/ModelGenerators.scala]: can we update the tests so they do some testing?

Always putting None here is leaving the writing and parsing unchecked. I'd like to have a proper generator here.

File: test/scala/com/github/johnynek/bazel_deps/ParseTest.scala [hunk 2]
```diff
@@ -106,6 +107,7 @@ class ParseTest extends FunSuite {
               None,
               None,
               None,
+              None,
```
Comment (johnynek) [inline test/scala/com/github/johnynek/bazel_deps/ParseTest.scala]: can you add some examples that do have licenses?

**Comment by johnynek** [comment]
Thanks so much for the PR! I'm really excited to merge!

File: test/scala/com/github/johnynek/bazel_deps/ParseTest.scala (hunk 2 above)
Comment (paladine) [reply test/scala/com/github/johnynek/bazel_deps/ParseTest.scala]: Added to "parse a file with options, yaml"

File: test/scala/com/github/johnynek/bazel_deps/ModelGenerators.scala (hunk 1 above)
Comment (paladine) [reply test/scala/com/github/johnynek/bazel_deps/ModelGenerators.scala]: I wasn't 100% sure how to add a test and validate against the final BUILD output. Looks like a lot of the testing is autogenerated?

I did replace None with a random set of licenses though

**Comment by johnynek** [comment]
Great! Thanks!

**Pull request merged**